main.py

```diff
@@ -0,0 +1,153 @@
+import asyncio
+import logging
+import os
+from asyncio import sleep
+from datetime import datetime
+from functools import wraps
+from open_ai_pressentation_explainer.pptx_processor import PptxProcessor
+
+import openai
+from flask.cli import load_dotenv
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+                    filename='info.log')
+ROLE = "user"
+GENERATION_TIMEOUT = 10  # seconds to wait for a response from OpenAI
+PATH = 'sources/asyncio-intro.pptx'
+ERROR_OCCURRED_MESSAGE = "An error occurred while processing the presentation:"
+LOG_FILE_PATH = 'info.log'
+RATE_LIMIT_SECONDS = 1
+WAIT_TIME = 60
+RATE_LIMIT_MSG = f"Rate limit exceeded. Please wait a {WAIT_TIME} seconds and try again."

```

Comment:
Very good use of constants!
Since you have so many constants, I would even create a file named "config" and put all of these constants there